specifyweb/sp7_models/models.py

```diff
@@ -0,0 +1,101 @@
+    timestampmodified = models.DateTimeField(blank=True, null=True, unique=False, db_column='TimestampModified', db_index=False, default=timezone.now)
+    text1 = models.TextField(blank=True, null=True, unique=False, db_column='Text1', db_index=False)
+    text2 = models.TextField(blank=True, null=True, unique=False, db_column='Text2', db_index=False)
+    text3 = models.TextField(blank=True, null=True, unique=False, db_column='Text3', db_index=False)
+
+    # Relationships: Many-to-One
+    collection = models.ForeignKey('specify.Collection', db_column='CollectionID', related_name='collectionobjectgroups', null=False, on_delete=protect_with_blockers)
+    createdbyagent = models.ForeignKey('specify.Agent', db_column='CreatedByAgentID', related_name='+', null=True, on_delete=protect_with_blockers)
+    modifiedbyagent = models.ForeignKey('specify.Agent', db_column='ModifiedByAgentID', related_name='+', null=True, on_delete=protect_with_blockers)
+
+    class Meta:
+        db_table = 'collectionobjectgroup'
+        ordering = ()
+
+    timestamptracker = FieldTracker(fields=['timestampcreated', 'timestampmodified'])
+    save = partialmethod(custom_save)
+
+class CollectionObjectGroupJoin(models.Model): # aka. CoJo or CogJoin
+    specify_model = datamodel.get_table('collectionobjectgroupjoin')
+
+    # ID Field
+    id = models.AutoField(primary_key=True, db_column='collectionobjectgroupjoinid')
+
+    # Fields
+    # TODO: Check if order, version, timestampcreated, and timestampmodified are needed
+    isprimary = models.BooleanField(blank=True, null=True, unique=False, db_column='IsPrimary', db_index=False)
+    issubstrate = models.BooleanField(blank=True, null=True, unique=False, db_column='IsSubstrate', db_index=False)
+    precedence = models.SmallIntegerField(blank=True, null=True, unique=False, db_column='Precedence', db_index=False)
+    order = models.SmallIntegerField(blank=True, null=True, unique=False, db_column='Order', db_index=False)
+    version = models.IntegerField(blank=True, null=True, unique=False, db_column='Version', db_index=False, default=0)
+    timestampcreated = models.DateTimeField(blank=False, null=False, unique=False, db_column='TimestampCreated', db_index=False, default=timezone.now)
+    timestampmodified = models.DateTimeField(blank=True, null=True, unique=False, db_column='TimestampModified', db_index=False, default=timezone.now)
+    text1 = models.TextField(blank=True, null=True, unique=False, db_column='Text1', db_index=False)
+    text2 = models.TextField(blank=True, null=True, unique=False, db_column='Text2', db_index=False)
+    text3 = models.TextField(blank=True, null=True, unique=False, db_column='Text3', db_index=False)
+
+    # Relationships: Many-to-One
+    collectionobjectgroupparent = models.ForeignKey('CollectionObjectGroup', db_column='CollectionObjectGroupParentID', related_name='cojos', null=False, on_delete=models.CASCADE)
```

Comment:
hmm I don't see value in being so verbose. Rename it to `collectionobjectgroupparent` to `groupparent`?